### DISCUSSION TITLE
Quarkus multi-module example with JPA and REST Client

### DISCUSSION THREAD

**Discussion by faermanj** [quarkusio/quarkus#31484] [Q&A] [answered]
Hello,

We're trying to set a very simple example of how to get a quarkus multi-module application working, including REST api and client, with a single entity (Greeting).

We have tried several different configuration alternatives and workarounds, but still not able to get a simple rest API/JPA call across to the http client module. 

Here is the repo:
https://github.com/CaravanaCloud/quarkus-multimodule-example

At the time of this writing, CDI is erroring because the EM is not found, because the data module only contains JPA annotations:
```
[ERROR]         [error]: Build step io.quarkus.arc.deployment.ArcProcessor#validate threw an exception: javax.enterprise.inject.spi.DeploymentException: javax.enterprise.inject.UnsatisfiedResolutionException: Unsatisfied dependency for type javax.persistence.EntityManager and qualifiers [@Default]
[ERROR]         - java member: org.acme.GreetingResource#em
```

We would appreciate any input towards making this example correct. 

Thanks @gsmet and everyone in advance for helping out <3

[a comment by an automated account is not shown]

**Reply by Sgitario** [reply]
The problem is that Quarkus does not detect any entities and hence it's disabling the `quarkus-hibernate-orm` dependency (that's why the `EntityManager` bean is not resolved). This is because Quarkus does not automatically inspect all the dependencies from your classpath. 
So, if you want your `data` module (the modules containing your entities) to be "seen" by your `api` module, you need to add the following plugin in the `data` module:

```
<build>
        <plugins>
            <!-- The entity classes need to be indexed -->
            <plugin>
                <groupId>io.smallrye</groupId>
                <artifactId>jandex-maven-plugin</artifactId>
                <executions>
                    <execution>
                        <id>make-index</id>
                        <goals>
                            <goal>jandex</goal>
                        </goals>
                    </execution>
                </executions>
            </plugin>
        </plugins>
    </build>
```

And now, when building your `api` module, the entities will be discovered and you can inject the EntityManager bean.

**Reply by gsmet** [reply]
Yes, that's why I try to tell @faermanj here: https://twitter.com/gsmet_/status/1630630472781840393 but Twitter is kinda limited for that :).

I created a PR here that should make things work: https://github.com/CaravanaCloud/quarkus-multimodule-example/pull/1

**Reply by faermanj** [reply]
Thank you, i will try that out right away.

**Comment by faermanj** [comment]
That worked, thanks a lot <3
Not sure where i missed that, perhaps here?
https://quarkus.io/guides/hibernate-orm#defining-entities-in-external-projects-or-jars
I'd be glad to update that if desirable...

**Reply by faermanj** [reply]
Is there any practical difference between adding beans.xml or the maven plugin?
